tests/e2e/specs/in-game/spectating.spec.js

```diff
@@ -360,3 +366,75 @@ describe('Spectating Games', () => {
     });
   });
 });
+
+describe('Creating And Updating Unranked Matches With Rematch - Spectating', () => {
+  beforeEach(function () {
+    cy.wipeDatabase();
+    cy.visit('/');
+
+    // Sign up players
+    cy.signupOpponent(playerOne).as('playerOneId');
+    cy.signupOpponent(playerThree).as('playerThreeId'); // spectator
+    // Opponent will be player 2 (the last one we log in as)
+    cy.signupOpponent(playerTwo).as('playerTwoId');
+
+    // Log in as playerOne
+    cy.loginPlayer(playerOne);
+    cy.setupGameAsSpectator();
+  });
+  it('Spectate unranked games with rematch', function () {
+    // 1st game: Opponent concedes
+    cy.recoverSessionOpponent(playerTwo);
+    cy.concedeOpponent();
+    assertVictory();
+    cy.log('rematch player2');
+
+    cy.rematchOpponent({ rematch: true });
+
+    cy.get('[data-cy="opponent-wants-rematch"]').should('be.visible');
+
+    cy.window()
+      .its('cuttle.gameStore')
+      .then((game) => {
+        cy.expect(game.p0Rematch).to.be.null;
+        cy.expect(game.p1Rematch).to.be.true;
+      });
+
+    cy.recoverSessionOpponent(playerOne);
+
+    cy.wait(1000);
+
+    cy.rematchOpponent({ rematch: true });
+    cy.wait(1000);
```

Comment:
Todo: see if we can avoid these by improving the promise management of `cy.recoverSessionOpponent()`

Reply:
https://github.com/cuttle-cards/cuttle/issues/795